osidb/tests/factories.py

```diff
@@ -300,6 +300,9 @@ class Meta:
     )
     ps_module = factory.sequence(lambda n: f"ps-module-{n}")
     ps_component = factory.sequence(lambda n: f"ps-component-{n}")
+    purl = factory.sequence(
+        lambda n: f"pkg:rpm/fedora/ps-component-{n}@7.50.3-1.fc25?arch=i386&distro=fedora-25"
+    )
```

Comment:
It is not yet being validated whether the PS component actually corresponds to the PURL but then we must not forget to update this co the PS component and the PURL string corresponds and the factory generates a valid affect. I would suggest to address it already so it is not forgotten but leaving it up to you.

Reply:
Right now this actually holds true as when not specifying these fields will result in the correct affect. Once we add that validaton, there should be probably a purl generated first in the factory and ps_component added via `factory.LazyAttribute` which will parse out the component from the purl via the library.

Reply:
Or the other way around if we can generate the correct PURL that way - setting the actually provided PS component to the purl instead of `ps-component-{n}` so it corresponds. I am thinking about this way because there is already a number of tests which specify `ps_component` explicitly and all of those would start failing.

Reply:
That would be however a little bit counter-intuitive to the fact that we would like to give higher priority tu the purls compared to the ps_components. Then passing just the PURL will then fail as the ps_component will be generated differently and the validation will fail.

Reply:
It could probably be done somehow conditionally both ways. Or it the tests we can for now prioritize the other way as there are tests with component and no PURL and not the other way.